src/Managers/DataManager.py

```diff
@@ -48,14 +48,15 @@ def get_file_path(self, file_chooser: Any) -> Any:
     def show_data_grid(self, grid_class: Any, output_handler: Any) -> None:
         """Show the file data grid."""
         output_handler.clear_output(wait=True)
-
+        number_of_lines = len(self._data.file)-1
+        magic_line_constant = 10000 
         with output_handler:
             if self._data.file is None:
                 print("Please, upload the file first!\u274C")
                 return
-
-            df = pd.DataFrame(data=self._data.file, columns=self._data.headers)
-            dg = grid_class(dataframe=df)
+            if number_of_lines>magic_line_constant:
+                print('Warning: File is too large ({0} rows). The output was truncated to {1} rows'.format(number_of_lines,magic_line_constant))
+            dg = grid_class(dataframe=self._data.file.iloc[:min(magic_line_constant,number_of_lines)])
```

Comment:
Tan Lei had some thoughts about that, as I remember, so we should probably leave it as it is for now.